velox/benchmarks_large/TpchBenchmark.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+#include <gflags/gflags.h>
+
+#include "velox/common/file/FileSystems.h"
+#include "velox/connectors/hive/HiveConnector.h"
+#include "velox/dwio/common/Options.h"
+#include "velox/dwio/parquet/reader/ParquetReader.h"
+#include "velox/exec/PlanNodeStats.h"
+#include "velox/exec/Split.h"
+#include "velox/exec/tests/utils/HiveConnectorTestBase.h"
+#include "velox/exec/tests/utils/TpchQueryBuilder.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+using namespace facebook::velox::dwio::common;
+
+namespace {
+
+static bool notEmpty(const char* /*flagName*/, const std::string& value) {
+  return !value.empty();
+}
+} // namespace
+
+DEFINE_string(data_path, "", "Root path of TPC-H data");
+DEFINE_int32(query, 1, "Query number");
+DEFINE_int32(num_drivers, 4, "Number of drivers");
+DEFINE_string(format, "parquet", "Data format");
+DEFINE_bool(verbose, false, "Print benchmark results in detail");
+
+DEFINE_validator(data_path, &notEmpty);
+
+class TpchBenchmark {
+ public:
+  void initialize() {
+    functions::prestosql::registerAllScalarFunctions();
+    filesystems::registerLocalFileSystem();
+    parquet::registerParquetReaderFactory();
+    dwrf::registerDwrfReaderFactory();
+    auto hiveConnector =
+        connector::getConnectorFactory(
+            connector::hive::HiveConnectorFactory::kHiveConnectorName)
+            ->newConnector(kHiveConnectorId, nullptr);
+    connector::registerConnector(hiveConnector);
+  }
+
+  std::shared_ptr<Task> run(const TpchPlan& tpchPlan) {
+    CursorParameters params;
+    params.maxDrivers = FLAGS_num_drivers;
+    params.numResultDrivers = 1;
+    params.planNode = tpchPlan.plan;
+    constexpr int kNumSplits = 10;
```

Comment:
This is number of splits per file, not total number of splits, right? If so, consider renaming to kNumSplitsPerFile for clarity. Would it make sense to make this configurable?

Reply:
will fix.